src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -32,12 +32,15 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
 
             properties.Headers = message.Headers.ToDictionary(p => p.Key, p => (object)p.Value);
 
-            if (message.Headers.ContainsKey(NServiceBus.Headers.EnclosedMessageTypes))
+            string messageTypesHeader;
+            if (message.Headers.TryGetValue(NServiceBus.Headers.EnclosedMessageTypes, out messageTypesHeader))
             {
-                properties.Type = message.Headers[NServiceBus.Headers.EnclosedMessageTypes].Split(new[]
+                var index = messageTypesHeader.IndexOf(',');
+
+                if (index > 0)
                 {
-                    ','
-                }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+                    properties.Type = messageTypesHeader.Substring(0, index);
+                }
```

Comment:
Strictly speaking, yes this is a change, but it's one that I can't see impacting us, since the EnclosedMessageTypes header contains the AssemblyQualifiedName of the type, which is going to have a comma.

I could tweak this a bit to something like:

```csharp
if (index > -1)
{
    properties.Type = messageTypesHeader.Substring(0, index);
}
else
{
     properties.Type = messageTypesHeader
}
```

which would match match for the case you mention. If the string had a leading comma, the behavior would still be different. 

However, given that we're working with AssemblyQualifiedNames, does that matter?